--- feeds/luci/libs/nixio/src/file.c	2014-03-03 16:37:08.218620266 +0100
+++ feeds/luci/libs/nixio_a/src/file.c	2014-03-03 17:50:30.450482433 +0100
@@ -223,6 +223,26 @@
 	}
 }
 
+#ifdef __linux__
+
+static int nixio_file_numexp(lua_State *L) {
+	int fd = nixio__checkfd(L, 1);
+	uint64_t numexp;
+	int readc;
+
+	do {
+		readc = read(fd, &numexp, sizeof(uint64_t));
+	} while (readc == -1 && errno == EINTR);
+
+	if (readc < 0) {
+		return nixio__perror(L);
+	} else {
+		lua_pushnumber(L, (lua_Number)numexp);
+		return 1;
+	}
+}
+
+#endif
 
 static int nixio_file_seek(lua_State *L) {
 	int fd = nixio__checkfd(L, 1);
@@ -350,6 +370,9 @@
 static const luaL_reg M[] = {
 	{"write",		nixio_file_write},
 	{"read",		nixio_file_read},
+#ifdef __linux__
+	{"numexp",		nixio_file_numexp},
+#endif
 	{"tell",		nixio_file_tell},
 	{"seek",		nixio_file_seek},
 	{"stat",		nixio_file_stat},
